### PR TITLE
Added gingko test for e2e

### PR DESCRIPTION
**Pull request by aneeshkp**
Signed-off-by: Aneesh Puttur <aneeshputtur@gmail.com>
Added basic framework to add gingko test
Next PR will add all the functionality.
To run functest 
`make funcstest`

### CODE REVIEW

File: test/cne/cne.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package cne
```
Comment (jzding) [inline test/cne/cne.go:1]: The package name does not match what's in the Makefile:
`SUITE=./test/cloud-event-proxy `

Comment (aneeshkp) [reply test/cne/cne.go:1]: Thanks, Fixed

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -26,6 +26,11 @@ jobs:
     steps:
     - name: Checkout code
       uses: actions/checkout@v2
+    - uses: actions/setup-go@v2
+      with:
+        go-version: '1.17'
+   - name: tidy
+     run: go mod tidy
```
Comment (jzding) [inline .github/workflows/main.yaml:33]: Not sure if this is an issue but these 2 lines do not line up.

Comment (aneeshkp) [reply .github/workflows/main.yaml]: Fixed

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -26,6 +26,11 @@ jobs:
     steps:
     - name: Checkout code
       uses: actions/checkout@v2
+    - uses: actions/setup-go@v2
+      with:
+        go-version: '1.17'
+    - name: tidy
+     run: go mod tidy
```
Comment (jzding) [inline .github/workflows/main.yaml:33]: this line not line up

**Review by jzding — APPROVED** [review]

**Pull request merged**